src/OpenLoco/src/GameCommands/VehicleReverse.cpp

```diff
@@ -0,0 +1,58 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Types.hpp"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004ADAA8
+    static uint32_t vehicleReverse(EntityId headId, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        auto* head = EntityManager::get<Vehicles::VehicleHead>(headId);
+        if (head == nullptr)
+        {
+            return FAILURE;
+        }
+
+        if (!sub_431E6A(head->owner))
+        {
+            return FAILURE;
+        }
+
+        auto* head2 = EntityManager::get<Vehicles::VehicleHead>(head->head);
```

Comment:
I would probably just skip this. It looks like this function was original designed to be called with not a head but any vehicle part but since we know its all heads we can just skip this.

Reply:
sorry i meant remove the getting the head again. Keep the canBeModified check

Reply:
Oh right, that makes sense. Didn't think about that, will revert that part.